subsys/bluetooth/controller/hci/hci.c

```diff
@@ -5253,6 +5271,91 @@ static void le_ext_adv_report(struct pdu_data *pdu_data,
 	adv_info->length = data_len;
 	memcpy(&adv_info->data[0], data, data_len);
 
+	if (!scan_data) {
+		node_rx_extra_list_release(node_rx->hdr.rx_ftr.extra);
+
+		return;
+	}
+
+	/* Allocate, append as buf fragement and construct the scan response
+	 * event.
+	 */
+	struct net_buf *scan_evt_buf = bt_buf_get_rx(BT_BUF_EVT, K_FOREVER);
+
+	net_buf_frag_add(buf, scan_evt_buf);
+
+	/* If scan data complete */
+	if (!scan_data_status) {
+		/* Only copy data that fit the event buffer size,
+		 * mark it as incomplete
+		 */
+		if (scan_data_len > data_max_len) {
+			scan_data_len = data_max_len;
+			scan_data_status =
+				BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_PARTIAL;
+		}
+
+	/* else, data incomplete */
+	} else {
+		/* Only copy data that fit the event buffer size */
+		if (scan_data_len > data_max_len) {
+			scan_data_len = data_max_len;
+		}
+	}
+
+	/* Start constructing the event */
+	info_len = sizeof(struct bt_hci_evt_le_ext_advertising_info) +
+		   scan_data_len;
+	sep = meta_evt(scan_evt_buf, BT_HCI_EVT_LE_EXT_ADVERTISING_REPORT,
+		       sizeof(*sep) + info_len);
+
+	sep->num_reports = 1U;
+	adv_info = (void *)(((uint8_t *)sep) + sizeof(*sep));
+
+	/* Set scan response bit */
+	evt_type |= BT_HCI_LE_ADV_EVT_TYPE_SCAN_RSP;
+
+	/* set scan data status bits */
+	evt_type |= (scan_data_status << 5);
+
+	adv_info->evt_type = evt_type;
+
+	if (0) {
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	} else if (rl_idx < ll_rl_size_get()) {
+		/* Store identity address */
+		ll_rl_id_addr_get(rl_idx, &adv_info->addr.type,
+				  &adv_info->addr.a.val[0]);
+		/* Mark it as identity address from RPA (0x02, 0x03) */
+		adv_info->addr.type += 2U;
+#endif /* CONFIG_BT_CTLR_PRIVACY */
+	} else if (adv_addr) {
+		adv_info->addr.type = adv_addr_type;
+		memcpy(&adv_info->addr.a.val[0], adv_addr, sizeof(bt_addr_t));
+	} else {
+		adv_info->addr.type = 0U;
+		memset(&adv_info->addr.a.val[0], 0, sizeof(bt_addr_t));
+	}
+
+	adv_info->prim_phy = find_lsb_set(phy);
+	adv_info->sec_phy = sec_phy;
+	adv_info->sid = (adi) ? adi->sid : BT_HCI_LE_EXT_ADV_SID_INVALID;
+	adv_info->tx_power = tx_pwr;
+	adv_info->rssi = rssi;
+	adv_info->interval = interval_le16;
+
+	if (evt_type & BT_HCI_LE_ADV_EVT_TYPE_DIRECT) {
+		adv_info->direct_addr.type = direct_addr_type;
+		memcpy(&adv_info->direct_addr.a.val[0], direct_addr,
+		       sizeof(bt_addr_t));
```

Comment:
Are you typically casting unused return values to `void` in the controller? If so, consider doing so for `memcpy` and `memset` as well (i.e. `(void)memcpy(...)`.